feature-libs/quote/components/details/overview/quote-details-overview.component.ts

```diff
@@ -5,50 +5,259 @@
+  iconTypes = ICON_TYPE;
+  editMode = false;
 
   constructor(
     protected quoteFacade: QuoteFacade,
+    protected eventService: EventService,
     protected translationService: TranslationService
   ) {}
 
-  //TODO: consider to create similar generic function for all cx-card usages
-  getCardContent(value: string | null, titleKey: string): Observable<Card> {
-    return this.translationService.translate(titleKey).pipe(
-      map((title) => ({
-        title,
-        text: [value ?? '-'],
-      }))
+  protected defineQuoteMetaData(event: EditEvent): QuoteMetadata {
+    const [name, description, expirationTime] = [
+      event.name,
+      event.description,
+      event.expirationTime,
+    ];
+
+    let metaData: QuoteMetadata = {};
+    metaData = {
+      ...metaData,
+      ...(event.name && { name }),
+      ...(event.description && { description }),
+      ...(event.expirationTime && { expirationTime }),
+    };
+
+    return metaData;
+  }
+
+  /**
+   * Verifies whether the quote information card tile is editable.
+   *
+   * @param {Quote} quote - quote
+   * @returns {boolean} - if the quote is editable and its state is 'QuoteState.BUYER_DRAFT' or 'QuoteState.BUYER_OFFER', otherwise returns 'false'.
+   */
+  isQuoteInformationEditable(quote: Quote): boolean {
+    return (
+      quote.isEditable &&
+      (quote.state === QuoteState.BUYER_DRAFT ||
+        quote.state === QuoteState.BUYER_OFFER)
     );
   }
+
+  /**
+   * Cancels the view of the edit card tile
+   * by setting the edit mode to 'false'.
+   *
+   * @param {boolean} isEditMode - edit event
+   */
+  cancel(isEditMode: boolean) {
+    this.editMode = isEditMode;
+  }
+
+  /**
+   * Edits the card tile.
+   *
+   * @param {Quote} quote - Quote
+   * @param {EditEvent} event - edit event
+   */
+  edit(quote: Quote, event: EditEvent) {
+    this.editMode = event.editMode;
+    const metaData: QuoteMetadata = this.defineQuoteMetaData(event);
+
+    this.quoteFacade
+      .editQuote(quote.code, metaData)
+      .pipe(take(1))
+      .subscribe(
+        // success
+        () => {
+          this.eventService.dispatch({}, QuoteDetailsReloadQueryEvent);
+        }
+      );
+  }
+
+  /**
+   * Sets the edit mode to the opposite.
+   */
+  setEditMode() {
+    this.editMode = !this.editMode;
+  }
+
+  /**
+   * Retrieves the card content that represents the quote information with its name and description.
+   *
+   * @param {string} name - Quote name
+   * @param {string} description - Quote description
+   * @returns {Observable<Card>} - Card content
+   */
+  getQuoteInformation(name?: string, description?: string): Observable<Card> {
+    return combineLatest([
+      this.translationService.translate('quote.details.information'),
+      this.translationService.translate('quote.details.name'),
+      this.translationService.translate('quote.details.description'),
+    ]).pipe(
+      map(([infoTitle, nameTitle, descriptionTitle]) => {
+        return {
```

Comment:
We don't need {return ....} but can directly map to the result

Reply:
Discussed!